.github/workflows/checks-quick.yml

```diff
@@ -135,11 +135,285 @@ jobs:
+
+          # Complex version formats
+          test_full_pipeline "Complex version format" '{"packages":[{"name":"polkadot-sdk","version":"4.0.0-rc1"}]}' "false" "" "polkadot-v4.0" "4.0.0-rc1" && \
+
+          # Feature branch scenarios  
+          test_full_pipeline "Feature branch with metadata" '{"packages":[{"name":"polkadot-sdk","version":"9.9.9"}]}' "false" "" "feature/something" "9.9.9" && \
+          test_full_pipeline "Feature branch fallback" '{}' "false" "" "wolfenheimm/check-umbrella-crate" "0.1.0" && \
+
+          # Multiple packages in metadata
+          test_full_pipeline "Multiple packages" '{"packages":[{"name":"other-pkg","version":"1.0.0"},{"name":"polkadot-sdk","version":"3.0.7"}]}' "false" "" "polkadot-v3.0" "3.0.7"
+
+          full_test_result=$?
+
+          echo -e "\n=== Test results ==="
+          if [ $branch_test_result -eq 0 ] && [ $full_test_result -eq 0 ]; then
+            echo "All tests passed!"
+          else
+            echo "Some tests failed!"
+            exit 1
+          fi
+
+          # Reset any mocked functions
+          unset -f git
+          unset -f cargo
+          unset -f command
+          unset -f jq
+          unset -f test
+          unset -f grep
       - name: check umbrella correctness
         run: |
           # Fixes "detected dubious ownership" error in the ci
           git config --global --add safe.directory '*'
-          python3 scripts/generate-umbrella.py --sdk . --version 0.1.0
+
+          # Ensure jq is installed
+          if ! command -v jq &> /dev/null; then
+            echo "Installing jq..."
+            apt-get update && apt-get install -y jq
+          fi
+
+          # Extract the umbrella crate version dynamically from cargo metadata
+          # This handles both master and stable branches correctly
+          UMBRELLA_VERSION=""
+
+          # First attempt: get version from cargo metadata
+          UMBRELLA_VERSION=$(cargo metadata --format-version=1 | jq -r '.packages[] | select(.name == "polkadot-sdk") | .version')
+
+          # Second attempt: check Cargo.toml if exists
+          if [ -z "$UMBRELLA_VERSION" ] && [ -d "umbrella" ] && [ -f "umbrella/Cargo.toml" ]; then
+            UMBRELLA_VERSION=$(grep -m 1 'version = ' umbrella/Cargo.toml | cut -d '"' -f 2)
+          fi
+
+          # Last resort: fallback to branch-based version or 0.1.0
+          if [ -z "$UMBRELLA_VERSION" ]; then
+            CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+            if [[ "$CURRENT_BRANCH" =~ ^polkadot-v([0-9]+\.[0-9]+)$ ]] || [[ "$CURRENT_BRANCH" =~ ^release-v([0-9]+\.[0-9]+)$ ]]; then
+              # For stable branches, use the branch version with .0 suffix
+              BRANCH_VERSION=$(echo "$CURRENT_BRANCH" | grep -oE 'v[0-9]+\.[0-9]+' | sed 's/v//')
+              UMBRELLA_VERSION="${BRANCH_VERSION}.0"
+              echo "Using version ${UMBRELLA_VERSION} derived from branch name ${CURRENT_BRANCH}"
```

Comment:
We can remove these later attempts. It should be fine relying only on `cargo metadata`.